generators/app/index.js

```diff
@@ -27,65 +33,76 @@ module.exports = Generator.extend({
         name: 'githubUsername',
         message: 'What is the github project organisation or username?',
         validate: required,
-        default: githubUsername
+        default: githubUsername,
+        when: utils.noConfig('githubUsername', this.initialConfig)
```

Comment:
As the options are inputted via the command line they should all be stringified, although this probably isn't the most robust assumption.

How about using this module: https://www.npmjs.com/package/validate.io-undefined-or-null

and then `when: isUndefinedOrNull(this.initialConfig['githubUsername'])`